lib/Component.lua

```diff
@@ -424,4 +431,33 @@ function Component:_unmount()
 	self._handle = nil
 end
 
+--[[
+	Performs property validation, if it is enabled.
+]]
+function Component:_validateProps(props)
+	if not GlobalConfig.getValue("propertyValidation") then return end
+
+	local validator = self.validateProps
+
+	if validator == nil then return end
+	if typeof(validator) ~= "function" then
+		-- Hide as much as possible about error location, since this message
+		-- occurs from several possible call sites with different stack traces.
+		-- luacheck: ignore 6
+		error(("The value of validateProps must be a function (is a %q). Check the definition of the component extended at:\n%s"):format(
+			typeof(validator),
+			self._extendTraceback),
+		0)
+	end
+
+	local success, failureReason = validator(props)
+
+	if not success then
+		failureReason = failureReason or "<No failure reason was given by the validation function>"
+		error(("Property validation failed:\n%s\n%s"):format(
+			failureReason,
```

Comment:
We should explicitly call `tostring` on the failure reason in case the validator wants to express more detailed errors in some cases.